native/src/fairseq2n/data/round_robin_data_source.cc

```diff
@@ -18,11 +18,12 @@ round_robin_data_source::round_robin_data_source(
 {
     buffer_.reserve(pipelines_.size());
 
-    is_infinite_ = std::all_of(
-        pipelines_.begin(), pipelines_.end(), [](const data_pipeline &p)
+    finitude_type_ = pipelines_.empty() ? data_source_finitude_type::finite : std::max_element(
```

Comment:
Nit: Same as above. I would split this into 2 or 3 separate statements for better readability.